frontend/document/documentregistry.lua

```diff
@@ -56,9 +56,17 @@ end
 function DocumentRegistry:hasProvider(file)
     local filename_suffix = string.lower(util.getFileNameSuffix(file))
 
-    if self.filetype_provider[filename_suffix] then
+    local filetype_provider = G_reader_settings:readSetting("provider") or {}
+    if self.filetype_provider[filename_suffix] or filetype_provider[filename_suffix] then
         return true
     end
+    local DocSettings = require("docsettings")
+    if DocSettings:hasSidecarFile(file) then
+        local doc_settings_provider = DocSettings:open(file):readSetting("provider")
+        if doc_settings_provider then
+            return true
```

Comment:
Was going to suggest you should properly do a `docsetting:close()` like it is done in many other places (eg. https://github.com/koreader/koreader/pull/5384).
But no, letting the table disappear when out of scope, as done here, is the right thing to do.
These other places, by calling `:close()`, do unnecessarily flush/save the unmodified data to disk...